internal/mempool/mempool.go

```diff
@@ -2159,4 +2474,196 @@ func New(cfg *Config) *TxPool {
 		staged:          make(map[chainhash.Hash]*dcrutil.Tx),
 		stagedOutpoints: make(map[wire.OutPoint]*dcrutil.Tx),
 	}
+
+	mp.miningView = &txMiningView{
+		rejected:           make(map[chainhash.Hash]struct{}),
+		txGraph:            mp.newTxDescGraph(),
+		txDescs:            nil,
+		trackAncestorStats: cfg.Policy.EnableAncestorTracking,
+		ancestorStats:      make(map[chainhash.Hash]*mining.TxAncestorStats),
+	}
+
+	return mp
+}
+
+// TxDescs returns a collection of all transactions available in the view.
+func (mv *txMiningView) TxDescs() []*mining.TxDesc {
+	return mv.txDescs
+}
+
+// defaultAncestorStats is the default-state of ancestor stats for a transaction
+// that has no ancestors.
+var defaultAncestorStats = &mining.TxAncestorStats{}
+
+// AncestorStats returns the view's cached statistics for all of the provided
+// transaction's ancestors.
+func (mv *txMiningView) AncestorStats(txHash *chainhash.Hash) (*mining.TxAncestorStats, bool) {
+	if !mv.trackAncestorStats {
+		return defaultAncestorStats, false
+	}
+
+	if stats, exists := mv.ancestorStats[*txHash]; exists {
+		return stats, exists
+	}
+
+	return defaultAncestorStats, false
+}
+
+// Ancestors returns a collection of all transactions in the graph that the
+// provided transaction hash depends on, and its ancestors' bundle stats.
+func (mv *txMiningView) Ancestors(txHash *chainhash.Hash) []*mining.TxDesc {
+	var ancestors []*mining.TxDesc
+	if !mv.trackAncestorStats {
+		return nil
+	}
+
+	if mv.trackAncestorStats {
```

Comment:
This check is unnecessary since you are already checking `!mv.trackAncestorStats` and returning above

Reply:
Removed